marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/EntitySearchController.java

```diff
@@ -54,7 +57,22 @@ public String search(@RequestBody SearchQuery searchQuery, @RequestParam(default
     @RequestMapping(method = RequestMethod.GET)
     @ResponseBody
     public JsonNode getRecord(@RequestParam String docUri, @RequestParam(defaultValue = "final") String database) {
-        return getEntitySearchService(database).getRecord(docUri);
+        return getEntitySearchService(database).getRecord(docUri, true);
+    }
+
+    @RequestMapping(value = "/downloadRecord", method = RequestMethod.GET)
+    public void downloadRecord(@RequestParam String docUri, @RequestParam(defaultValue = "final") String database, HttpServletResponse response) {
+        String[] docUriArray = docUri.split("/");
+        String fileName = docUriArray[docUriArray.length-1];
+        response.setContentType(MediaType.APPLICATION_OCTET_STREAM_VALUE);
+        response.addHeader("Content-Disposition", String.format("attachment; filename=%s", fileName));
+
+        try (OutputStream out = response.getOutputStream()) {
+            newEntitySearchManager(database).downloadRecord(docUri, out);
```

Comment:
I think this can be simpler and wholly implemented in the controller by using a DocumentManager from a DatabaseClient. Just read the URI as an InputStreamHandle, then use Spring's FileCopyUtils class to copy that class's InputStream to the HTTP response's OutputStream. You can remove all the code from the Manager (which doesn't really add any value here), and only have one try/catch too.